src/Protocols.Abstractions/Features/IConnectionItemsFeature.cs

```diff
@@ -6,8 +6,8 @@
 
 namespace Microsoft.AspNetCore.Protocols.Features
 {
-    public interface IConnectionMetadataFeature
+    public interface IConnectionItemsFeature
     {
-        IDictionary<object, object> Metadata { get; set; }
```

Comment:
What's going to end up here? Why object -> object instead of string -> object?

Reply:
https://github.com/aspnet/HttpAbstractions/blob/fbbbb16305f57025eca71964ac5d7c9f43c42150/src/Microsoft.AspNetCore.Http.Features/IItemsFeature.cs Also string keys are inferior.

Reply:
I mean strings are just objects after all. I should be embracing the fact we're using objects for keys. It should serve as a bright neon sign telling developers just to create their own features for anything important.

Reply:
Strings are slower keys (due to the fact the whole value of them must be compared)

Reply:
Yappp!

Reply:
@benaadams surely that depends on the efficiency of the key's Equals and GetHashCode methods.

I concede that you could create an object type that's a faster key than string most of the time. Though I wonder if string *does* end up being used as the object key, is it just as fast in a `Dictionary<object, object>` as it is in `Dictionary<string, object>`? Or does the extra compile-time info about the key result in more efficient IL?

Reply:
String in `object` would be slower as it would use randomised hashing; whereas `string` does a bait`n`switch to non-randomised hashing, switching back when collisions get high.

Reply:
Thinking about this more, if you're going to use the same object instance as your lookup value every time, strings don't make much sense. So I get your point there. Interesting info about the string hashing. I'll have to take a look at that.

Reply:
[public Dictionary(int capacity, IEqualityComparer<TKey> comparer)](https://github.com/dotnet/coreclr/blob/ffc16020fee4d866de568042fe085143332a2159/src/mscorlib/shared/System/Collections/Generic/Dictionary.cs#L80-L84)
```csharp
if (typeof(TKey) == typeof(string) && _comparer == null)
{
    // To start, move off default comparer for string which is randomised
    _comparer = (IEqualityComparer<TKey>)NonRandomizedStringEqualityComparer.Default;
}
```